src/OpenTracingActiveRecordBehavior.php

```diff
@@ -0,0 +1,165 @@
+            ->setTag('db.active_record.primary_key', json_encode($this->getOwner()->getPrimaryKey()));
+        $this->activeScope->close();
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onBeforeDelete} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * You may set {@link CModelEvent::isValid} to be false to quit the deletion process.
+     * @param CEvent $event event parameter
+     */
+    public function beforeDelete($event)
+    {
+        if(!$this->traceDelete && \Yii::app()->hasComponent($this->opentracing)) {
+            return;
+        }
+
+        $options = $this->spanOptionsFromActiveRecordClass();
+        $options['tags']['db.active_record.primary_key'] = json_encode($this->getOwner()->getPrimaryKey());
+
+        $this->activeScope = \Yii::app()->hasComponent($this->opentracing)
+            ->startActiveSpan($this->spanName('DELETE'), $options);
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onAfterDelete} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function afterDelete($event)
+    {
+        if(!$this->traceDelete && !$this->activeScope) {
+            return;
+        }
+        $this->activeScope->close();
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onBeforeFind} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function beforeFind($event)
+    {
+        if(!$this->traceFind && !\Yii::app()->hasComponent($this->opentracing)) {
+            return;
+        }
+
+        $this->activeScope = \Yii::app()->hasComponent($this->opentracing)
+            ->startActiveSpan($this->spanName('FIND'), $this->spanOptionsFromActiveRecordClass());
+    }
+
+    /**
+     * Responds to {@link CActiveRecord::onAfterFind} event.
+     * Override this method and make it public if you want to handle the corresponding event
+     * of the {@link CBehavior::owner owner}.
+     * @param CEvent $event event parameter
+     */
+    public function afterFind($event)
+    {
+        if(!$this->traceFind && !$this->activeScope) {
+            return;
+        }
+
+        $this->activeScope->getSpan()->log(['criteria' => $this->getOwner()->getDbCriteria()->toArray()]);
+        $this->activeScope->close();
+    }
+
+    private function spanName($action)
+    {
+        return sprintf('DB-AR::%s %s', strtoupper($action), get_class($this->getOwner()));
+    }
+
+    private function spanOptionsFromActiveRecordClass()
+    {
+        return [
+            'tags' => [
+                \OpenTracing\Tags\COMPONENT => 'yii-opentracing-activerecord',
```

Comment:
if you want to override the default one, please change it to `yii-opentracing.activerecord`

Reply:
changed